locations/commands/nsi.py

```diff
@@ -60,6 +69,30 @@ def lookup_code(self, args):
                 )
                 print("       -> " + str(item))
 
+    def detect_missing(self, args):
+        codes = {}
+        for spider_name in self.crawler_process.spider_loader.list():
+            props = DuplicateWikidataCommand.spider_properties(spider_name)
+            for code in props["wikidata_codes"]:
+                code_spiders = codes.get(code, set())
+                code_spiders.add(props["filename"])
+                codes[code] = code_spiders
+
+        # Fetch the category from NSI's github, and try to match to wikidata.
+        # TODO: This assumes you are going for only one category, by wikidata ID.
+        #       Is it worth having this just check all of the wikidata entries and printing out what is missing globally?
+        response = self.nsi._request_file(f"data/{args[0]}.json")
```

Comment:
I tried this.

Unfortunately nsi.min.json doesn't have the categorisation (shop/xyz), so you end up with generating everything in no particular order.
```
 [{'displayName': '1-2-3', 'id': '123-175913', 'locationSet': {'include': ['dk', 'pl']}, 'tags': {'brand': '1-2-3', 'brand:wikidata': 'Q4545742', 'advertising': 'totem'}, 'fromTemplate': True}, {'displayName': '7-Eleven', 'id': '7eleven-445a61', 'locationSet': {'include': ['001'], 'exclude': ['pl']}, 'matchNames': ['7-11', 'seven eleven'], 'tags': {'brand': '7-Eleven', 'brand:wikidata': 'Q259340', 'advertising': 'totem'}, 'fromTemplate': True}, {'displayName': '76', 'id': '76-445a61', 'locationSet': {'include': ['001'], 'exclude': ['pl']}, 'tags': {'brand': '76', 'brand:wikidata': 'Q1658320', 'advertising': 'totem'}, 'fromTemplate': True}, {'displayName': '8 à Huit', 'id': '8ahuit-2974c2', 'locationSet': {'include': ['fr']}, 'tags': {'brand': '8 à Huit', 'brand:wikidata': 'Q2818601', 'advertising': 'totem'}, 'fromTemplate': True}, {'displayName': 'ABC', 'id': 'abc-99bd2f', 'locationSet': {'include': ['fi']}, 'tags': {'brand': 'ABC', 'brand:wikidata': 'Q10397504', 'advertising': 'totem'}, 'fromTemplate': True}, {'displayName': 'Aegean', 'id': 'aegean-d67647', 'locationSet': {'include': ['gr']}, 'tags': {'brand': 'Aegean', 'brand:wikidata': 'Q28146598', 'advertising': 'totem'}, 'fromTemplate': True}, {'displayName': 'Afriquia', 'id': 'afriquia-445a61', 'locationSet': {'include': ['001'], 'exclude': ['pl']}, 'tags': {'brand': 'Afriquia', 'brand:wikidata': 'Q2829178', 'advertising': 'totem'}, 'fromTemplate': True}, {'displayName': 'Afriquia أفريقيا', 'id': '3331ce-3f0b55', 'locationSet': {'include': ['ma']}, 'tags': {'brand': 'Afriquia أفريقيا', 'brand:wikidata': 'Q56300032', 'name:ar': 'أفريقيا', 'advertising': 'totem'}, 'fromTemplate': True}, {'displayName': 'Agip', 'id': 'agip-98293a', 'locationSet': {'include': ['001'], 'exclude': ['ch', 'it']}, 'tags': {'brand': 'Agip', 'brand:wikidata': 'Q377915', 'advertising': 'totem'}, 'fromTemplate': True}
 ```

Reply:
https://raw.githubusercontent.com/osmlab/name-suggestion-index/main/dist/presets/nsi-id-presets.json is slightly closer, but focused strongly on OSM tagging rather than brand/xyz structures. I will experiment with this in a separate branch.

Reply:
Issue 2:
Checking the whole NSI - and not being able to filter to just brand/shop type entries results in excessive amount of output - ~600 MB vs ~5mb of just aggregating all of the brand/shop/* results.

Reply:
https://github.com/CloCkWeRX/alltheplaces/tree/whole-nsi for that version, but for now, backing away from that approach.

Reply:
Unless I'm missing something, it does. They are in the keys eg `.nsi."brands/advertising/totem".items[0]`, and also the tags.

Reply:
That'll teach me to try to parse JSON with my eyeballs...
I'll have anoher look then.